src/main/java/games/strategy/engine/lobby/server/db/UserController.java

```diff
@@ -96,6 +98,27 @@ public void updateUser(final DBUser user, final HashedPassword hashedPassword) {
   }
 
   @Override
+  public void updateUser(final DBUser user, final String password) {
+    Preconditions.checkArgument(user.isValid(), user.getValidationErrorMessage());
+
+    try (final Connection con = connectionSupplier.get()) {
+      try (final PreparedStatement ps =
+          con.prepareStatement("update ta_users set password = ?,  email = ?, admin = ? where username = ?")) {
+        ps.setString(1, BCrypt.hashpw(password, BCrypt.gensalt()));
```

Comment:
With just one line different, it does not seem the while method should be copy pasted compared to the hashed password version. Most straight forward way I see is create a simple enum. But really that is wrong too. `HashedPassed` does not mean `md5` hashed, it means that the password is no longer plain text, that is *has* been hashed. By adding the new APIs we are taking the responsibility of when the password is hashed from the caller, and moving it here.  This is not obvious for the casual observer who sees two methods  `updateUser` with slightly different APIs that seem to be identical. This is particularly so given `HashedPassword` is just an instance of a strongly typed `String`. I believe that means the right way to fix this is to move the conditional password logic to the time when `HashedPassword` is created, before we call these methods.